maps/~mapsystem/maps.dm

```diff
@@ -249,11 +249,18 @@ var/global/const/MAP_HAS_RANK = 2		//Rank system, also togglable
 		if(world.maxy < planet_size[2])
 			world.maxy = planet_size[2]
 	for(var/i = 0, i < num_exoplanets, i++)
-		var/exoplanet_type = pick(subtypesof(/obj/effect/overmap/visitable/sector/exoplanet))
+		var/exoplanet_type = pick_exoplanet()
 		INCREMENT_WORLD_Z_SIZE
 		var/obj/effect/overmap/visitable/sector/exoplanet/new_planet = new exoplanet_type(null, world.maxz)
 		new_planet.build_level(planet_size[1], planet_size[2])
 
+/datum/map/proc/pick_exoplanet()
+	var/planets = list()
+	for(var/T in subtypesof(/obj/effect/overmap/visitable/sector/exoplanet))
```

Comment:
Maybe cache this on /datum/map so it doesn't have to pick a bunch of times?

Reply:
it's just 2-3 times during server init, dunno if worth optimizing like that